draft-lmbdhk-quic-multipath.md

```diff
@@ -517,45 +518,111 @@ not cause linkability issue.
+the client and the server use non-zero length CIDs. Path identifier type 0x00
+is chosen. In this case, we are going to close the first path. For the first
+path, the server's 1-RTT packets use DCID C1, which has a sequence number of 1;
+the client's 1-RTT packets use DCID S2, which has a sequence number of 2.  For
+the second path, the server's 1-RTT packets use DCID C2, which has a sequence
+number of 2; the client's 1-RTT packets use CID S3, which has a sequence number
+of 3. Note that two paths use different packet number spaces.
 
 ~~~
-  Client                                                          Server
+  Client                                                      Server
 
   (client tells server to abandon a path)
-  1-RTT[X]: DCID=S2 PATH_ABANDON[path_id=1]->
-                                 (server tells client to abandon a path)
-        <-1-RTT[Y]: DCID=C1 PATH_ABANDON[path_id=2], ACK_MP[Seq=2, PN=X]
+  1-RTT[X]: DCID=S2 PATH_ABANDON[path_id_type=0, path_id=1]->
+                              (server tells client to abandon a path)
+          <-1-RTT[Y]: DCID=C1 PATH_ABANDON[path_id_type=0,path_id=2],
+                                                  ACK_MP[Seq=2, PN=X]
   (client abandons the path that it is using)
-  1-RTT[U]: DCID=S3 RETIRE_CONNECTION_ID[2], ACK_MP[Seq=1, PN=Y] ->
-                             (server abandons the path that it is using)
-       <- 1-RTT[V]: DCID=C2 RETIRE_CONNECTION_ID[1], ACK_MP[Seq=3, PN=U]
+  1-RTT[U]: DCID=S3 RETIRE_CONNECTION_ID[2],
+                          ACK_MP[Seq=1, PN=Y]->
+                          (server abandons the path that it is using)
+                         <-1-RTT[V]: DCID=C2 RETIRE_CONNECTION_ID[1],
+                                                  ACK_MP[Seq=3, PN=U]
 
 ~~~
-{: #fig-example-path-close title="Example of closing a path (path id type=0x00)"}
+{: #fig-example-path-close1 title="Example of closing a path when both the
+server and client use non-zero length CIDs"}
+
+{{fig-example-path-close2}} illustrates an example of path closing when the
+client uses non-zero length CID while the server uses zero-length CID. Similar
+to the above example, in this case, we are going to close the first path. For
+the first path, the client's 1-RTT packets use DCID S2, which has a sequence
+number of 2, so the server can still use path identifier type 0x00 and a
+path_id=2 in its PATH_ABANDON frame. However, because the server now uses
+zero-length CID, the client needs to use path identifier type 0x01 with a
+path_id=2 in its PATH_ABANDON frame, which is different from the above example.
+The server can readily close the first path when it receives the
+RETIRE_CONNECTION_ID frame from the client. However, since the client will not
+receive a RETIRE_CONNECTION_ID frame, after sending out RETIRE_CONNECTION_ID
+frame, the client waits for 3 RTO before closing the path. Note that single
+packet number spaces are used in both directions.
```

Comment:
Isn't that always the case that if a single packet number space is used in one directly is also needs to be used in the other direction?

Also the assumption here is because there is a non-zero length CID in on direction, this has to use one packet number space? Maybe better so say this at the beginning of this paragraph.